src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.hpp

```diff
@@ -0,0 +1,426 @@
+   */
+  virtual void initializePML() override;
+
+  /**
+   * @brief Apply Perfectly Matched Layer (PML) to the regions defined in the geometry box from the xml
+   * @param time the time to apply the BC
+   * @param domain the partition domain
+   */
+  virtual void applyPML( real64 const time, DomainPartition & domain ) override;
+
+
+  /// save the sismo trace in file
+  void saveSeismo( localIndex iseismo, real64 val, string const & filename ) override;
+
+  /// Indices of the nodes (in the right order) for each source point
+  array2d< localIndex > m_sourceNodeIds;
+
+
+  /// Constant part of the source for the nodes listed in m_sourceNodeIds in x-direction
+  array2d< real64 > m_sourceConstantsx;
+
+  /// Constant part of the source for the nodes listed in m_sourceNodeIds in x-direction
+  array2d< real64 > m_sourceConstantsy;
+
+  /// Constant part of the source for the nodes listed in m_sourceNodeIds in x-direction
+  array2d< real64 > m_sourceConstantsz;
+
+  /// Flag that indicates whether the source is local or not to the MPI rank
+  array1d< localIndex > m_sourceIsLocal;
+
+  /// Indices of the element nodes (in the right order) for each receiver point
+  array2d< localIndex > m_receiverNodeIds;
+
+  /// Basis function evaluated at the receiver for the nodes listed in m_receiverNodeIds
+  array2d< real64 > m_receiverConstants;
+
+  /// Flag that indicates whether the receiver is local or not to the MPI rank
+  array1d< localIndex > m_receiverIsLocal;
+
+  /// Displacement_np1 at the receiver location for each time step for each receiver (x-component)
+  array2d< real64 > m_displacementxNp1AtReceivers;
+
+  /// Displacement_np1 at the receiver location for each time step for each receiver (y-component)
+  array2d< real64 > m_displacementyNp1AtReceivers;
+
+  /// Displacement_np1 at the receiver location for each time step for each receiver (z-component)
+  array2d< real64 > m_displacementzNp1AtReceivers;
+
+};
+
+
+namespace extrinsicMeshData
+{
+
+EXTRINSIC_MESH_DATA_TRAIT( Displacementx_nm1,
+                           "displacementx_nm1",
+                           array1d< real64 >,
+                           0,
+                           NOPLOT,
+                           WRITE_AND_READ,
+                           "x-component of displacement at time n-1." );
+
+EXTRINSIC_MESH_DATA_TRAIT( Displacementy_nm1,
+                           "displacementy_nm1",
+                           array1d< real64 >,
+                           0,
+                           NOPLOT,
+                           WRITE_AND_READ,
+                           "y-component of displacement at time n-1." );
+
+EXTRINSIC_MESH_DATA_TRAIT( Displacementz_nm1,
+                           "displacementz_nm1",
+                           array1d< real64 >,
+                           0,
+                           NOPLOT,
+                           WRITE_AND_READ,
+                           "z-component of displacement at time n-1." );
```

Comment:
For my own understanding, why did you use 3 `array1d` for displacement, instead of just one `array2d` storing the 3 components of displacement? I am asking because we already define a displacement variable in the solid mechanics solvers as an `array2d`. See here:
https://github.com/GEOSX/GEOSX/blob/36f108a19fec1ffb2429b4e7316c8bcb8fddd999/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp#L151-L155
so I just wonder if there is a good reason to do things differently in the elastic solver?

Reply:
Indeed there is not... I think when, I did the first version of the solver ( a long time ago) my understanding of GEOSX was very low so I take the easy way by putting three vectors as I did not think about the resizeDimension stuff
Since the first order now I know it so we use it^^'